betelgeuse.py

```diff
@@ -36,6 +37,28 @@
 
 JUNIT_TEST_STATUS = ['error', 'failure', 'skipped']
 
+# Cache for shared objects
+OBJ_CACHE = {}
+
+
+class JobNumberParamType(click.ParamType):
+    name = 'job number'
+
+    def convert(self, value, param, context):
+        if value == 'auto':
+            return multiprocessing.cpu_count()
+        try:
+            value = int(value)
+            if value <= 0:
+                raise ValueError('{0} is not a positive integer'.format(value))
+            return value
+        except ValueError:
```

Comment:
The try/except is getting errors when converting the string to integer and the error raised on line 53 is related to the number not being positive. They are two different things.